lib/src/platform/src/rest/rest.dart

```diff
@@ -36,42 +32,56 @@ class Rest extends PlatformObject implements AblyBase {
     return handle;
   }
 
-  @override
-  Auth? auth;
+  /// a custom auth object to perform authentication related operations
+  /// based on the [options]
+  ///
+  /// https://docs.ably.com/client-lib-development-guide/features/#RSC5
+  // Auth? auth;
 
-  @override
-  ClientOptions options;
+  /// [ClientOptions] indicating authentication and other settings for this
+  /// instance to interact with ably service
+  late ClientOptions options;
 
-  @override
-  Future<HttpPaginatedResponse> request({
-    String? method,
-    String? path,
-    Map<String, dynamic>? params,
-    Object? body,
-    Map<String, String>? headers,
-  }) {
-    throw UnimplementedError();
-  }
+  /// creates and sends a raw request to ably service
+  ///
+  /// https://docs.ably.com/client-lib-development-guide/features/#RSC19
+  // Future<HttpPaginatedResponse> request({
+  //   String? method,
+  //   String? path,
+  //   Map<String, dynamic>? params,
+  //   Object? body,
+  //   Map<String, String>? headers,
+  // }) {
+  //   throw UnimplementedError();
+  // }
 
-  @override
-  Future<PaginatedResult<Stats>> stats([Map<String, dynamic>? params]) {
-    throw UnimplementedError();
-  }
+  /// gets stats based on params as a [PaginatedResult]
+  ///
+  /// https://docs.ably.com/client-lib-development-guide/features/#RSC6
+  // Future<PaginatedResult<Stats>> stats([Map<String, dynamic>? params]) {
+  //   throw UnimplementedError();
+  // }
 
-  @override
-  Future<DateTime> time() {
-    throw UnimplementedError();
-  }
+  /// returns server time
+  ///
+  /// https://docs.ably.com/client-lib-development-guide/features/#RSC16
+  // Future<DateTime> time() {
+  //   throw UnimplementedError();
+  // }
 
-  @override
+  /// a push object interacting with Push API, such as
+  /// subscribing for push notifications by clientId.
```

Comment:
Similar situation here. I'd leave the documentation , comment to the class implementation

Reply:
This documentation is for the SDK user, and this type (Rest) are being exposed to the user. The implementation is not exposed to the user, so they won't see the documentation.